File: 1. Two Sum.cc
// https://leetcode.com/problems/two-sum/description/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
     
        for (int i = 0; i < nums.size(); i++) {
            for (int j = nums.size() - 1; j > i; j--) {
                if (nums[i] + nums[j] == target) {
                    return {i, j};
                }
            }
        }
    }
};

File: 1. Two Sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        location = {}

        for i, n in enumerate(nums):
            if (target - n in location):
                return [i, location[target - n]]
            else:
            	location[n] = i

nums = [3,2,4,2,1,2,4,32,3,53,45,3]
target = 8

print(Solution().twoSum(nums, target))